src/goal.rs

```diff
@@ -785,11 +786,69 @@ impl<'a> Goal<'a> {
   }
 
   /// Consume this goal and add its case splits to the proof state
-  fn case_split(mut self, state: &mut ProofState<'a>) {
+  fn case_split(mut self, state: &mut ProofState<'a>) -> CaseSplitStatus {
     let new_lemmas = self.add_lemma_rewrites(state);
 
+    let mut blocking_vars: HashSet<_> = HashSet::default();
+
+    for reduction in self.reductions {
+      let x = reduction.searcher.get_pattern_ast().unwrap();
+      let sexp = symbolic_expressions::parser::parse_str(&x.to_string()).unwrap();
+
+      let mut new_sexps: Vec<Sexp> = self
+        .analyze_sexp_for_blocking_vars(&sexp)
+        .into_iter()
+        .map(|x| x.to_string())
+        .collect::<HashSet<_>>()
+        .into_iter()
+        .map(|x| symbolic_expressions::parser::parse_str(x.as_str()).unwrap())
+        .collect();
```

Comment:
This is a hack for deduping sexps, yes? Since they don't implement `PartialOrd` (and therefore we cannot use them in a tree-based set even).

Reply:
Yup